oracle/config/app.go

```diff
@@ -152,17 +152,33 @@ func ReadConfigFromAppOpts(opts servertypes.AppOptions) (AppConfig, error) {
 		}
 	}
 
+	if !cfg.Enabled {
+		return cfg, nil
+	}
+
 	// get the oracle address
 	if v := opts.Get(flagOracleAddress); v != nil {
-		if cfg.OracleAddress, err = cast.ToStringE(v); err != nil {
-			return cfg, err
+		address, err := cast.ToStringE(v)
+		if err != nil {
+			return cfg, fmt.Errorf("oracle address must be a non-empty string")
```

Comment:
this won't return an error in the empty case

Reply:
yea, we only replace if its non-empty. if its empty we just use the default.